cover_agent/CoverageProcessor.py

```diff
@@ -211,8 +211,18 @@ def parse_coverage_report_jacoco(self) -> Tuple[list, list, float]:
         total coverage percentage is returned to be evaluated only.
         """
         lines_covered, lines_missed = [], []
+        source_file_extension = self.get_file_extension(self.src_file_path)
+
+        package_name, class_name = "",""
+        if source_file_extension == 'java':
+            package_name, class_name = self.extract_package_and_class_java()
+        elif source_file_extension == 'kt':
+            package_name, class_name = self.extract_package_and_class_kotlin()
+
+        else:
+            raise ValueError(f"Unsupported Byte Class Language: {source_file_extension}")
```

Comment:
@red-velet 
Can you explain why are adding here a 'raise ValueError' ?

Won't it crash the run for any other language ?

Reply:
Currently, projects using the JaCoCo plugin primarily focus on Java and Kotlin, with other JVM family languages being relatively uncommon. If an exception is not raised here and the default Java/Kotlin parsing logic is silently applied to handle other JVM family languages, it may lead to the following issues:

1. **User Awareness Issue**: CoverAgent users would be unaware of this default behavior and may mistakenly believe that other languages are being handled correctly.

2. **Coverage Calculation Errors**: Since the JaCoCo report format is designed based on Java/Kotlin's structure, applying Java's parsing logic to other JVM family languages could result in incorrect coverage calculations or even 0% coverage. This could mislead users about the actual coverage and impact the validity of the tests.

For this reason, raising an exception explicitly alerts users that only Java and Kotlin are supported, preventing potential misuse and related issues. However, I also recognize that throwing an exception would terminate the program, potentially affecting user experience. Therefore, it would be more user-friendly to log a warning instead, informing users that the language is unsupported, while continuing with the default logic. This approach retains compatibility and clearly communicates potential risks to users.

**Modified Code Example:**

```python
source_file_extension = self.get_source_file_extension(self.src_file_path)

if source_file_extension == '.kt':
    package_name, class_name = self.extract_package_and_class_kotlin()
elif source_file_extension == '.java':
    package_name, class_name = self.extract_package_and_class_java()
else:
    # Log a warning using the logger
    self.logger.warn(f"Unsupported Bytecode Language: {source_file_extension}. Using default Java logic.")
    package_name, class_name = self.extract_package_and_class_java()
```

**Explanation of Changes:**
1. Replaced `raise ValueError` with `self.logger.warn()` to log a warning rather than terminating the program.
2. Retained `extract_package_and_class_java` as the fallback logic, ensuring the program continues to function.
3. Used logging instead of `print`, aligning with best practices in production environments and enabling centralized log management.

This change strikes a balance between user experience and robustness while paving the way for future support of additional JVM family languages.

How does this approach look to you?☺

Reply:
"jacoco" is for java. got it 👍